src/transformers/models/bert/modeling_bert.py

```diff
@@ -417,36 +440,52 @@ def forward(
         head_mask=None,
         encoder_hidden_states=None,
         encoder_attention_mask=None,
+        past_key_value=None,
         output_attentions=False,
     ):
+        self_attn_past_key_value = past_key_value[:2] if past_key_value is not None else None
```

Comment:
Also please add some comments here as in: https://github.com/huggingface/transformers/blob/ec07da65e25562040581febaf9b400a462962961/src/transformers/models/bart/modeling_bart.py#L433

the caching mechanism is by nature pretty complex IMO so it'd be nice to have very clean code and some helpful comments here